base/compiler/typelattice.jl

```diff
@@ -321,7 +322,8 @@ function stupdate!(state::Nothing, changes::StateUpdate)
         changeid = slot_id(changes.var::Slot)
         newst[changeid] = changes.vtype
         # remove any Conditional for this Slot from the vtable
-        for i = 1:length(newst)
+        # (unless this change is came from the conditional)
+        !changes.conditional && for i = 1:length(newst)
```

Comment:
I use this strange code style in order to minimize the diff, but I'm welcome to use the basic `if` block style if you want.

Reply:
Yeah, could you change this to standard style? I'll usually hide whitespace after I verify formatting, to examine the functional diff

Reply:
Sure thing. Change to the standard style, and also rebased against the current master.